### PR TITLE
fix(core): get v8 console from context extra bindings

### PR DESCRIPTION
**Pull request by legendecas**
Fix https://github.com/denoland/deno/issues/14658

### CODE REVIEW

File: core/bindings.rs [hunk 1]
```diff
@@ -131,6 +131,11 @@ pub fn initialize_context<'s>(
   // Bind functions to Deno.core.*
   set_func(scope, core_obj, "callConsole", call_console);
 
+  // Bind extra binding object to Deno.core.extra
+  let extra_str = v8::String::new(scope, "extra").unwrap();
+  let extra_binding_obj = context.get_extras_binding_object(scope);
+  core_obj.set(scope, extra_str.into(), extra_binding_obj.into());
```
Comment (bartlomieju) [inline core/bindings.rs:137]: Could we extract `console` prop here and bind it directly to `Deno.core.console` instead of adding new `extra` namespace?

**Comment by bartlomieju** [comment]
Thanks for looking into this @legendecas!

File: core/bindings.rs (hunk 1 above)
Comment (legendecas) [reply core/bindings.rs:137]: Yeah, we can do it. 

However, exposing the extra binding object directly allows us to facilitate additional extra bindings like trace APIs: https://source.chromium.org/chromium/chromium/src/+/main:v8/src/init/bootstrapper.cc;l=6127;drc=7e7414b2ce1f1ccb9e8b5ba578528d97abd6c6ad;bpv=0;bpt=1.

Comment (bartlomieju) [reply core/bindings.rs:137]: That sounds good for the future, but we don't have a burning need for it right now - in general we're moving towards removal of `Deno.core` API (#16881) so these APIs shouldn't be exposed to users as they are provided by V8. Would you mind opening an issue with proposal for tracing API?

Comment (legendecas) [reply core/bindings.rs]: I don't have any immediate need to expose the trace API. Updated with extracting the console object instead.

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks @legendecas!

**Pull request merged**